### PR TITLE
Adding actions to send single emails with body and template.

### LINKED ISSUE #3966: SMTP2Go: send email action
**Is there a specific app this action is for?**
SMTP2GO

**Please provide a link to the relevant API docs for the specific service / operation.**
https://apidoc.smtp2go.com/documentation/#/POST%20/email/send

### PR DESCRIPTION
**Pull request by bsommardahl**
Fixes #3966

I took a lot of hints from the Postmark app and actions to build out these two actions. I don't really use SMTP2GO, but I wanted to help out!

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4144"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/smtp2go/actions/send-single-email-with-template/send-single-email-with-template.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    ...common.props,
+  },
+  async run({ $ }) {
+    const data = {
+      ...this.getActionRequestCommonData(),
+      template_id: this.templateId,
+      template_data: this.templateModel
+    };
+    const response = await this.smtp2go.sendSingleEmailWithTemplate($, data, this.ignoreFailures);
+    $.export("$summary", "Sent email successfully");
```
Comment (lcaresia) [inline components/smtp2go/actions/send-single-email-with-template/send-single-email-with-template.ts:37]: You can update this in other actions too.

Obs: I didn't tested this.

```suggestion
    $.export("$summary", `Sent email successfully with id ${response.id}`);
```

**Review by lcaresia — CHANGES_REQUESTED** [review]
Thanks for your contribution!

Just one change resquested!

**Review by lcaresia — APPROVED** [review]

**Comment by vunguyenhung** [comment]
Hello @bsommardahl, first of all thank you for you contribution!

Though I couldn't register account on SMTP2GO to test this component end-to-end, are you happen to have a test account on SMTP2GO that I can use?
FireShot Capture 521 - SMTP2GO - SMTP Service Pricing   Signup - www smtp2go com

**Comment by bsommardahl** [comment]
@vunguyenhung how about if we ask the OP for the connected issue? I do not have a test account.

**Comment by vunguyenhung** [comment]
Hello @dannyroosevelt, for the context, I could not create SMTP2GO account to test end-to-end flow. After sending an email to SMTP2GO customer support, I got the response back saying that Turing domain was suspended a year ago due to previous usage, also the Customer Support Agent does not reply on the test account request from me.

I would suggest to merge this and monitor for error usage. What do you think?

**Comment by dannyroosevelt** [comment]
> Hello @dannyroosevelt, for the context, I could not create SMTP2GO account to test end-to-end flow. After sending an email to SMTP2GO customer support, I got the response back saying that Turing domain was suspended a year ago due to previous usage, also the Customer Support Agent does not reply on the test account request from me.
> 
> I would suggest to merge this and monitor for error usage. What do you think?

Sounds good, Leo

**Pull request merged**